sysvabi64/sysvabi64.rst

```diff
@@ -916,10 +923,639 @@ These follow the AAELF64_ base definition from Arm.
+private data, thus making the addresses available without compromising
+the position independence and shareability of a program's text
+segment. A program references its global offset table using
+position independent addressing and extracts the absolute values from
+it, thereby redirecting position independent references to their
+actual locations.
+
+The global offset table (GOT) is created by the static linker in
+response to GOT generating relocations. See AAELF64_ Relocation
+operations for more information.
+
+AArch64 splits the global offset table (GOT) into two sections:
+
+ * ``.got.plt`` for code addresses accessed only from the Procedure Linkage
+   Table (PLT).
+
+ * ``.got`` all other addresses and offsets.
+
+Both the ``.got`` and ``.got.plt`` sections are aligned to a minimum
+of 8 bytes. They both contain 8-byte sized entries.
+
+If a program requires direct access to the absolute address of a
+symbol, that symbol will have a GOT entry. Because the executable and
+shared objects have their own independent GOTs, a symbol's address may
+appear in several tables. The dynamic linker processes all of the
+relocations in the ``.got`` section before giving control to any code
+in the process image, thus ensuring the absolute addresses are
+available during execution.
+
+By convention the ``.got`` section's first entry (number zero) is
+reserved to hold the address of the dynamic structure, referenced with
+the linker defined symbol ``_DYNAMIC``. Some dynamic linker's use this
+address or another reserved symbol to find its own dynamic structure
+without without having yet processed their own relocation entries. For
+AArch64 entries one and two in the ``.got`` are also reserved.
+
+For AArch64 the linker defined ``_GLOBAL_OFFSET_TABLE_`` symbol should
+be the address of the first global offset table entry in the ``.got``
+section.
+
+See `Sample code sequences for code models`_ for code model for
+examples of how to access the ``.got``.
+
+Function Addresses
+^^^^^^^^^^^^^^^^^^
+
+Direct function calls are those where the name of the called function
+is known at compile time. The PC-relative BL instruction may be used
+for all direct function calls, whether absolute or position
+independent.
+
+Indirect function calls are those where the address of the function is
+in a pointer. Appropriate code is used to load the value of the
+pointer into a register, as for other data, and then a BLR instruction
+is used.
+
+In statically linked code the address of a function is always its real
+address. However in dynamically linked environments references to the
+address of a function from an executable file and its shared objects
+might not resolve to the same value.
+
+References from within shared objects to a function address will
+normally be resolved by the dynamic linker to the virtual address of
+the function itself.
+
+References from within a non position independent executable file to a
```

Comment:
ISTM `non` is always used as `non-foo` or `nonfoo`, but not `non foo`.

Reply:
I'd probably need to make it `non-position-dependent` rather than `non-position dependent`, will need to update all `position independent` to make it `position-independent`. Sadly the ABI as a whole is a little bit inconsitent in which form it has used.